File: src/main/java/io/myosotisdev/utopianism/modules/guild/Guild.java
package io.myosotisdev.utopianism.modules.guild;

import io.myosotisdev.utopianism.modules.region.IRegionKeeper;
import io.myosotisdev.utopianism.modules.region.Region;
import io.myosotisdev.utopianism.modules.player.ImplPlayer;

import java.util.HashMap;
import java.util.List;
import java.util.Set;
import java.util.UUID;

public class Guild implements IRegionKeeper
{
    private String       name;
    private List<String> manifest;
    private String       inventoryIconTextures;
    private UUID uid;
    private HashMap<UUID, ImplPlayer> mappedPlayers;

    private Set<Region> regions;
    private int level;

    public String getName()
    {
        return name;
    }

    public void setName(String name)
    {
        this.name = name;
    }

    public List<String> getManifest()
    {
        return manifest;
    }

    public void setManifest(List<String> manifest)
    {
        this.manifest = manifest;
    }

    public String getInventoryIconTextures()
    {
        return inventoryIconTextures;
    }

    public void setInventoryIconTextures(String inventoryIconTextures)
    {
        this.inventoryIconTextures = inventoryIconTextures;
    }

    public HashMap<UUID, ImplPlayer> getPlayers()
    {
        return mappedPlayers;
    }

    public Set<Region> getRegions()
    {
        return regions;
    }

    public UUID getUid()
    {
        return uid;
    }

    public void setUid(UUID uid)
    {
        this.uid = uid;
    }

    public int getLevel()
    {
        return level;
    }

    public void setLevel(int level)
    {
        this.level = level;
    }
}

File: src/main/java/io/myosotisdev/utopianism/modules/guild/Guild.kt
package io.myosotisdev.utopianism.modules.guild

import com.google.gson.JsonObject
import com.google.gson.JsonParser
import io.myosotisdev.minestom.Minestom
import io.myosotisdev.utopianism.modules.Economy
import io.myosotisdev.utopianism.modules.region.IRegionKeeper
import io.myosotisdev.utopianism.modules.region.Region
import net.minestom.server.entity.Player
import java.io.File
import java.io.FileReader
import java.util.*
import kotlin.collections.HashMap
import kotlin.collections.HashSet

class Guild : IRegionKeeper
{
    var name: String? = null;
    var manifest: MutableList<String>? = null
    var inventoryIconTextures: String? = null
    var leader: Player
    val players: HashMap<Player, GuildClass> = HashMap()
    override val regions: HashSet<Region> = HashSet()

    override fun acquireRegion(region: Region?)
    {
        TODO("Not yet implemented")
    }

    override fun releaseRegion(region: Region?)
    {
        TODO("Not yet implemented")
    }

    val uid: UUID
    var level = 0
    val bank: Economy = Economy()

    constructor(uid: UUID, name: String, player: Player)
    {
        this.uid = uid;
        this.name = name;
        this.leader = player
    }

    constructor(uid: UUID, gFile: File)
    {
        this.uid = uid;

        var gObject: JsonObject = JsonParser.parseReader(FileReader(gFile)).asJsonObject
        var jManifest = gObject["manifest"].asJsonArray
        var jPlayers = gObject["players"].asJsonArray
        var jRegions = gObject["regions"].asJsonArray

        this.name = gObject["name"].asString;
        this.manifest = MutableList(jManifest.size()) { index -> jManifest.get(index).asString }
        this.leader = Minestom.getPlayer(UUID.fromString(gObject["leader"].asString))!!
        this.level = gObject["level"].asInt
        this.bank.money = gObject["bank"].asDouble

        jPlayers.forEach { element ->
            var pObj = element.asJsonObject
            players.put(Minestom.getPlayer(UUID.fromString(pObj["uuid"].asString))!!, GuildClass.valueOf(pObj["class"].asString.uppercase()))
        }

        jRegions.forEach { element ->

        }
    }

    fun upgrade(): Boolean
    {
        return false
    }

    fun dissolve()
    {

    }
}